### ISSUE TITLE
[Question] Is there a way to include sub package dependency changes when using conventional changelog?

### ISSUE THREAD

**Issue by jmaher409** [lerna/lerna#1257]
When generating a changelog, I would like to have my public npm package changelogs include changes that were made to private packages that are in the same mono repo. For example, suppose a lerna mono repo contains the following packages:

packages/
    public-package1
    private-package1
    private-package2

and they have the following dependency relationship:

 public-package1 -> public-package1 -> public-package2

With the default behavior on publish (with conventional commits option) the public-package1 changelog only contains changes from that package and not from private-package1 and private-package2 packages. Those package are included in the bundle that public-package1 ships so I would like my consumers to be able to see the changelogs for all those dependencies. Is this possible? Thanks for any help!

**Comment by evocateur** [comment]
I agree that the current message for "dependency bump only" with `--conventional-commits` is unfortunate, I don't know that bubbling up all the changes is really appropriate. Changelogs are meant to describe changes relevant to the package in question, and conflating transitive changes seems to violate that goal.

**Comment by jmaher409** [comment]
Thanks for the response @evocateur. I completely agree that changes should describe only the package in question. For our particular use case, we use the mono repo to organize code within private packages inside our repo. There are 20+ packages but only 3 are actually published to npm. Some of the private packages contain code that use imported in the public ones. I would like changes from those packages to to appear in the public packages. I think this still adheres to the requirement that changelogs reflect the relevant package changes. Another option, of course, would be to also publish the internal packages so that those changelogs can be discovered. However that seems like it is forcing us to expose the existence of the internal packages even though the fact that they are there is an implementation detail.

**Comment by khaledosman** [comment]
have a similar need, in my case I have a mono-repo of 4 packages; two apps that rely on a shared UI components package as a design system as well as pure a Typescript SDK / Library.  

From my perspective it makes sense to include the sub packages changelog as part of the changelog / release notes of these consuming apps to reflect the changes to these components, I get the point that packages should include their own changes only, but when its used as "release notes" where these depending / consuming packages (the main apps) are not published it might make sense to include them, ideally it can be configurable from `lerna.json`

**Comment by ecnaidar** [comment]
Hey, were there any updates or workaround for this? 
Have same case as @khaledosman and wondering how should I proceed because many changes should be done in `common`. Such change actually seems to properly force version bump for "public package" but does not include any changes from 'common' in it's changelog

**Comment by dforesman** [comment]
_Throwing in my take on this in lieu of opening another issue thread:_

I would love to see the end of "version bump only" entries in changelogs when inter-dependent packages are published together.


# Example

For example, let's say I have a monorepo with `pkg-a`, and `pkg-b`. `pkg-b` includes `pkg-a` in its `dependencies`.

Let's say I add a feature to `pkg-a`, "feat: improves display of bolded text". On the next release, this will show up in `pkg-a`'s `CHANGELOG` under "Features", with the message "improves display of bolded text".

However, when `pkg-b` is auto-bumped during the next publish cycle, the same update in its changelog will only read "version bump only for `pkg-b`".


# Why is this a problem?

Among other things, this is a nightmare for understanding updates to packages that bundle other packages. At my company, we have many different tooling packages such as eslint config, jest config, etc, which are all included in a `build-tools` package.

Every single update published for `build-tools` _only_ includes the "Version bump only" text. Consumers have to manually dig into child packages for update details every time Dependabot pushes out an update to our teams. They don't even know which local dependency to look for changes in. This increases maintenance overhead and slows adoption of updates across our org.

<img width="1214" alt="changelog-example" src="https://user-images.githubusercontent.com/962283/194637031-386e0fcc-f252-4419-9a7c-2bc9c770752e.png">

While I understand that bubbling up _all_ transitive dependency updates would be beyond the scope of what Lerna is trying to provide, it seems sensible that it would be able to do so _specifically for local transitive deps in the same monorepo_. Perhaps optionally, to retain existing behavior for those that want it.

Lerna already makes assumptions about propagating version bumps through dependent packages, so it does not strike me as a fundamental change to add similar assumptions about propagating the description of _what those bumps do_.

# What might a solution look like?

**At minimum**, in the example scenario above, the entry for `pkg-b` should describe _what_ local dependency was version bumped, including the package name and the "to" versions. Example: "version bump `pkg-a` to `0.2.0`".

**Ideally**, since we have access to the commits that created the version bump, we could include those messages in the parent changelog as well. Example: "version bump `pkg-a` from `0.1.0` to `0.2.0`: improves display of bolded text"
